source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md

```diff
@@ -0,0 +1,147 @@
+* You probably need network access if you are going to run something like `buildah bud`. You don't need this if you are just running `buildah images/containers` etc. But many use cases will be using `bud` and so I've just used the hosts network and now the bridge because it's a short running build process. So for the example I'll use the `--network host` parameter for `podman run`.
+* You'll need to bind mount an area of the file system, so that `buildah bud` can build up and then commit the image to the host file system. This can be `/var/lib/containers` if you like, so as it’s immediately shared shared on the hose, or it can be some build sandbox area etc. You also need to bind mount the Dockerfile path so that Buildah can see the Dockerfile. So for the example I’ll use the bind mount parameter `-v` or `--volume` parameter for `podman run`.
+
+Use your favorite Nginx Dockerfile or the one below. 
+
+```
+# Base on the Fedora
+FROM fedora:latest
+MAINTAINER http://fedoraproject.org/wiki/Cloud
+
+RUN dnf -y update && dnf clean all
+RUN dnf -y install nginx && dnf clean all
+RUN echo "daemon off;" >> /etc/nginx/nginx.conf
+RUN echo "nginx on Fedora" > /usr/share/nginx/html/index.html
+
+EXPOSE 80
+
+ENTRYPOINT [ "/usr/sbin/nginx" ]
+```
+
+The following command builds an Nginx container image called ‘nginx’. Obviously using Podman (or Docker) is overkill because they both have build capabilities. Ideally this container would be run in a pod on kubernetes through CRI-O. My Dockerfile is in my `/home/whenry/dfs` directory (shorter for dockerfiles). So I bind mount that directory to `/tmp` inside the container. Inside it’s `/tmp/Dockerfile`.
+
+```
+# podman run --privileged --network host -v /var/lib/containers:/var/lib/containers:rw  -v /home/whenry/dfs:/tmp:Z buildah bud -f /tmp/Dockerfile -t nginx 
+```
+
+Currently, due to OverlayFS requiring root privileges, this container must run in privileged mode. This requirement should be deprecated when Fedora’s OverlayFS does not require a privileged user.
+
+## Current issues with non privileged mode
+
+There is a requirement in large clusters for buildah not to run in privileged mode. There are ways to mitigate this but there will be cases where people demand it.
+
+I had prepared an example of how to do this but I will keep it for a later blog.  Here are some of the current constraints I encountered trying to solve the non-privileged problem
```

Comment:
"non-privileged problem".  Not sure if you want an ':' or an '.' here, but I think you want one of the other at the end.

Reply:
+1 '.'